File: funcscript/parser/syntax/FuncScriptParser.GetExpression.cs
using FuncScript.Model;
namespace FuncScript.Core
{
    public partial class FuncScriptParser
    {
        static ExpressionBlockResult GetExpression(ParseContext context, int index)
        {
            var result = GetInfixExpression(context, index);
            if (result.NextIndex > index)
                return result;

            return new ExpressionBlockResult(null, null, index);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetExpression.ts
import { ParseContext, ParseResult } from "../FuncScriptParser.Main";
import { GetInfixExpression } from "./FuncScriptParser.GetInfixExpression";

export function GetExpression(context: ParseContext, index: number): ParseResult {
    const result = GetInfixExpression(context, index);
    if (result.NextIndex > index) {
        return result;
    }

    return { ParseNode: null, NextIndex: index };
}